File: subprogramas/linguagem C/params_valor_padrao.c

//Em C não podemos fazer assim como em C++, onde podemos passar os valores padrões dos parâmetros em sua declaração. Em vez disso, o que podemos fazer é utilizar macros ou definindo outras funções com um numero diferente de parâmetros.

#include <stdio.h>
#include <stdlib.h>

// função original com todos os parâmetros
void printValues(int a, int b) {
    printf("a = %d, b = %d\n", a, b);
}

// função que usa um valor padrão para o segundo parâmetro
void printValuesDefault(int a) {
    printValues(a, 10); // 10 é o valor padrão para b
}

File: subprogramas/linguagem PHP/params_valor_padrao.php

<?php

// Função original com todos os parâmetros
function printValues($a, $b = 10) {
    echo "a = $a, b = $b\n";
}

// Chamadas da função
printValues(5); // $b usará o valor padrão de 10
printValues(5, 20); // $b usará o valor passado de 20

?>
